src/core/server/http/http_server.ts

```diff
@@ -97,10 +99,11 @@ export class HttpServer {
     this.registeredRouters.add(router);
   }
 
-  public setup(config: HttpConfig): HttpServerSetup {
+  public async setup(config: HttpConfig): Promise<HttpServerSetup> {
     const serverOptions = getServerOptions(config);
     const listenerOptions = getListenerOptions(config);
     this.server = createServer(serverOptions, listenerOptions);
+    await this.server.register([HapiStaticFiles]);
```

Comment:
Would it make sense to specify this as part of the constructor arguments? https://hapi.dev/api/?v=17.9.2#-serveroptionsplugins

Though I'm not sure how that works with a promise/async plugin code?

Reply:
`server.options.plugins` allows adjusting plugin options. It doesn't register a plugin:
```
Plugin-specific configuration which can later be accessed via server.settings.plugins. plugins is an object where each key is a plugin name and the value is the configuration
```